### DISCUSSION TITLE
Digital Ziffern erkennung

### DISCUSSION THREAD

**Discussion by schnuller123** [jomjol/AI-on-the-edge-device#3452] [Q&A] [answered]
Hallo

ich habe Probleme einen Digitalen Wasserzähler auszuwerten
die hinteren Ziffern sind kleiner und haben einen Rand die das auswertfenster begrenzt
verstehe auch nicht ganz die erkennung,  was bedeutet zB 2.0 das er eine 2 erkannt hat beim ersten Versuch und was 9.9 ? nan ist klar
und was kann ich ändern? 
[
![Screenshot_20241225-153551_Chrome](https://github.com/user-attachments/assets/72f004be-6887-4730-8b42-17486a94c76e)
](url)

**Comment by SybexX** [comment]
na etwas mehr Informationen solltest du schon preisgeben, damit jemand dir helfen kann
- welche Version verwendest du
- wie sind deine Einstellungen
- was hast du schon versucht
- ...............

**Comment by schnuller123** [comment]
dachte im bild sieht man genügend
denke es ist ein problem der auswertfenster

v16.0.0 rc5
dig-cont_0712_s3_q.tflite

bild ausschnitt und schärfe sollte doch passen

versucht
kleiner oder grössere auswertfenster

als 1.tip was bedeutet 9.9 ?

**Comment by SybexX** [comment]
war ja ursprünglich für alte Wasserzähler mit Dreheziffern gedacht und da hast du ja ein Übergang von 9 auf 0.
Die digitalen Zähler zeigen nur ganze Zahlen(0, 1, 2, 3 .......), die mit Drehziffern können ja auf 9,  9.1,  9.2, ...... 9.9,  0,  0.1,  0.2 ....... stehen.

https://jomjol.github.io/AI-on-the-edge-device-docs/ROI-Configuration/
![Unbenannt](https://github.com/user-attachments/assets/c041ec57-7758-4140-a9f4-3c69d0199131)

Um aus der 9.9 eine 0 zu bekommen, könntest du auch versuchen, das Auswahlfenster etwas nach unten oder oben zu verschieben.

**Reply by schnuller123** [reply]
wow das war mal ein hinweis!
super und danke dafür
jetzt sieht es schon viel besser aus
verstand da bisher wenig davon, wusste zwar die antike vorgeschichte
aber jetzt haben doch viele einen Digitalen Zähler
wundere mich sowieso, das es nicht eine version gibt, die voll auf digitale ausgelegt ist

**Reply by Ingo-Knito** [reply]
Sind wir doch mal ehrlich: Es ist Unsinn, dieses Projekt auf digitale Zähler loszulassen.
Es ist eine geniale Idee für analoge Zähler und leistet dort einen ziemlich perfekten Job!
Aber bei einem digitalen Zähler müsste eine Kamera obsolet sein. Warum sind die Teile doch digital?!

Leider gibt es - zumindest nach meinen Recherchen - weltweit keinen einzigen Zähler, der seine Messdaten an einen Endverbraucher abliefert.
Es gibt einen Einzigen mit Funkverbindung. Die Funkverbindung basiert dort auf public-key Verfahren (super!). Allerdings braucht man dann den privaten Schlüssel. Den gibt aktuell leider kein (!) Wasserversorger an den Verbraucher heraus.
Das Teil ist - so nebenbei - nicht günstig...

**Reply by schnuller123** [reply]
voll auch meine Meinung
erster versuch war an den Key zu kommen
bin aber gescheitert
denke das haben wir unserem Datenschutz zu verdanken
vorteile nur für den wasserversorger
der fährt einfach die strasse ab und gut ist
frag mich auch wie lange der akku hält
lcd display und funk!
und alle 60s ein displaytest mit allen symbolen.....
epaper gibt es ja erst seit 10 jahren....

**Reply by schnuller123** [reply]
nochmal grosses Lob an das Forum
so sollte es immer sein
fachliche antwort und keine beleidigungen....

**Reply by Ingo-Knito** [reply]
Das mit dem Datenschutz wird meist vorgeschoben.
Hier ist es schlicht Investitionsschutz, da das Teil knapp 800 € kostet. Den Wasserversorger vllt 500 €. Da muss man viel Wasser verbrauchen, Bus sich dieser Invest rechnet.

**Comment by schnuller123** [comment]
Hallo nochmal eine Frage an alle

jetzt läuft es eigentlich sehr stabil, aber immer wieder passiert es,
das durch falsche erkennung der previous wert falsch definiert wurde
und deshalb richtige Werte nicht genommen werden, wegen dadurch erzeugte negative Ergebnisse.

Hat einer "AI on the Edge" am laufen mit Digitalem Zähler?
oder hat einer Tips um das Problem zu lösen

anbei ein logile und die config
[log.txt](https://github.com/user-attachments/files/18267541/log.txt)
[config.ini.txt](https://github.com/user-attachments/files/18267548/config.ini.txt)

wäre sehr toll wenn jemand eine lösung hat!

**Reply by SybexX** [reply]
Du könntest mit der Zoom-Funktion die Grösse der ROIs verändern, um die Erkennung zu verbessern.
Wenn sie zu groß oder zu klein sind, ist die Erkennung schlecht. Welche Grösse bei dir am besten ist, müsstest du noch rausfinden^^
Bei mir ist die Erkennung am besten, wenn ich dx auf 50 und dy auf 71 konfiguriert habe.
Und eventuell noch an den Kameraeinstellungen was verändern. Wenn das Auge dir sagt, dass das Bild gut ist, gilt das nicht unbedingt auch für den ESP.

**Comment by schnuller123** [comment]
problem ist auch noch das er alle 60s ein Displaytest macht und danach die Version 5.0 anzeigt

**Comment by schnuller123** [comment]
gute idee
aber was ist mit dem 888.888
vom displaytest?

**Comment by schnuller123** [comment]
und was macht die neue version anders?
dachte es gibt nun 3 marker

**Reply by schnuller123** [reply]
nun ist das bild viel zu hell
muss erstmal anpassen
ist das ok oder ein problem

**Reply by SybexX** [reply]
das was ich beschrieben habe, die "normale" Version hat keine Abbruchbedingungen beim Ausrichten (selbst wenn keine Ausrichtung zustande kommt, werden die "Werte" verwendet).

**Reply by SybexX** [reply]
Wichtig ist eine gute Trennung der Farben (kein fließender Farbverlauf), sieht fürs Auge vieleicht Schei.. aus, ist aber für die Auswertung besser.Wenn du die Überbelichtung durch Reduzierung der LED-Intensität nicht ausreichend reduzieren kannst, mußt du CamGainceiling von x8 auf x4 reduzieren.

**Reply by schnuller123** [reply]
hab ich schon gemacht x4 x2 x0
kein unterschied
auch agc probiert
alte config eingespielt und alles wieder ok, noch mit altem marker ausserhalb display
neuen marker auch h symbol 
reboot und wieder extrem hell

![Screenshot_20241228-234548_Chrome](https://github.com/user-attachments/assets/e5d9d9cb-c1db-4168-b223-583e8e3e9a50)

**Reply by SybexX** [reply]
hast du CamAec = true, CamAec2 = true ? bei false wird CamGainceiling nicht verwendet.

**Comment by schnuller123** [comment]
waren beide aus
hab auch mal grayscale probiert ohne erfolg
jetzt noch versuch mit 2x true

bingo!

aber warum nur probleme wenn marker im display?

**Comment by schnuller123** [comment]
Alignment failed
aber ziffern erkannt
etwas dunkel das bild
aber kein wert wegen aligment faild
kommt mit dem h nicht zurecht?
![Screenshot_20241228-235559_Chrome](https://github.com/user-attachments/assets/b32dfcdb-5d48-41c7-bd11-f6f64eb9c0ba)

**Reply by SybexX** [reply]
ja, eventuell "l/h" auswählen, oder das Bild etwas heller machen.

**Reply by SybexX** [reply]
hast du erst die LEDIntensity so weit wie möglich reduziert und dann erst CamGainceiling?

**Reply by schnuller123** [reply]
mach leds fest mit streifen und schalt relais mit der internen led spannung
also fällt aus!

**Reply by SybexX** [reply]
achso, also kannst du nur mit CamBrightness, CamContrast, CamSaturation und CamSharpness arbeiten^^

**Reply by schnuller123** [reply]
ok danke für hilfe, mach feierabend
morgen gehen die versuche weiter
danke extrem

**Comment by schnuller123** [comment]
so nun habe ich mit vorwiderstand die led abgeschwächt
und nun sieht es sehr gut aus mit der spezial Version von SybexX
danke nochmal für die Hilfe
werde in ein paar Tagen nochmal bescheid geben ob es nun OK ist!
![Screenshot_20241229-164314_Chrome](https://github.com/user-attachments/assets/d63a8fa4-05e8-465f-a973-f1d707bd9b35)

**Comment by schnuller123** [comment]
@SybexX  

sieht doch nich so gut aus, die 888.888 machen Probleme
frage:
könntest du eine Version machen wo ein 3. marker verwendet wird,
der nicht aktiv sein darf
aktiv ist dunkles feld innerhalb des markers

**Reply by SybexX** [reply, marked as answer]
Ist zu viel Aufwand und zusätzliche Belastung für den ESP32.
Meine Idee ist eher eine Liste von Werten, die ignoriert werden, ist einfacher/schneller umzusetzen und stellt kaum eine zusätzliche Belastung für den ESP32 dar.

**Reply by schnuller123** [reply]
und die neue macht sowas
888888 vorgabe wäre gut
aber wo einstellen?

**Reply by schnuller123** [reply]
ja habe ich, aber mit den problemen der helligkeit, hab ich es wieder lassen
aber das h ist doch genauso gut
problem ist jetzt nur noch die zahl 888888

**Reply by SybexX** [reply]
in der Version habe ich volgende rein gemacht: 
![Unbenannt](https://github.com/user-attachments/assets/7aac3bd3-60c7-4534-8345-a0c72d6efe3b)

Meine Überlegung ist, mehrere/bekannte Werte in der Firmware fest zu implementieren und einen Wert in den Einstellungen änderbar zu machen.

**Comment by schnuller123** [comment]
hier zb ein erfolg deine 4er version

![Screenshot_20241229-232130_Chrome](https://github.com/user-attachments/assets/71a75932-1625-4061-afa3-6bf712ede1f6)

dein vorschlag mit den festen werten ist gut, aber unflexibel 
ich habe kein komma, alles in liter
also ware für mich 8888 oder x888x besser
nur vorschlag, editieren im configfeld?
aber das hast fu ja eh vor...

**Reply by SybexX** [reply]
alle Zahlen sind doch dann 8 oder nicht? Das Komma wird ja durch den Dezimalshift hinzugefügt und muß im Display nicht vorhanden sein.

**Reply by schnuller123** [reply]
ich mach keine dezimalshift weil ich liter haben will
es sollte 888888 kommen

**Reply by schnuller123** [reply]
2024-12-29T23:08:23+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9
2024-12-29T23:12:16+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9
2024-12-29T23:13:28+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9
2024-12-29T23:14:40+0100,main,,,212409,,0,Alignment failed,N,N,5,0,N,N
2024-12-29T23:15:52+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9
2024-12-29T23:17:04+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9
2024-12-29T23:18:16+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9
2024-12-29T23:19:28+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9
2024-12-29T23:20:40+0100,main,,,212409,,0,Alignment failed,N,N,5,0,N,N
2024-12-29T23:21:52+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9
2024-12-29T23:23:04+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9
2024-12-29T23:24:16+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9
2024-12-29T23:25:28+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9
2024-12-29T23:26:40+0100,main,,,212409,,0,Alignment failed,N,N,5,0,N,N
2024-12-29T23:27:52+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9
2024-12-29T23:29:04+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9
2024-12-29T23:30:16+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9
2024-12-29T23:31:28+0100,main,212409,212409,212409,0.000000,0,no error,2,1,2,4,0,9

3x erfolg durch alligment not ok durch V4

**Reply by schnuller123** [reply]
sowas eben noch für 88888
wäre toll

**Reply by SybexX** [reply]
bei der 5 habe ich aus deinem log den Wert genommen und erweitert
![Unbenannt](https://github.com/user-attachments/assets/fb3ae807-20b3-4809-931b-ebc0951437e7)

**Reply by schnuller123** [reply]
versteh nicht was du meinst
die 4 löst doch schon das 1. problem aligment

**Reply by SybexX** [reply]
dies soll eine Alternative zum marker im Display sein^^ Sprich, wenn der Nutzer das mit den marker nicht weiß, wird der Wert trotzdem ignoriert.

**Reply by SybexX** [reply]
bei ver version 5 kannst du den Wert in der config.ini schon eintragen, jedoch noch nicht über die Konfig-Seite.
![Unbenannt](https://github.com/user-attachments/assets/9b7b2bd1-c080-466c-b65c-f7fb4c970c3c)

**Reply by schnuller123** [reply]
ok , ja klar verstanden
jetzt halt noch UserIgnorfeld für beliebige Zahl die ignoriert werden soll
super deine Arbeit und die Zusammenarbeit 
ich kenn das bei mir in der Firma etwas anders
Softwareentwickler sind immer etwas schwierig

so nun genug, geh ins bett und freue mich auf V6

gute nacht und Danke

**Reply by schnuller123** [reply]
du bist so schnell
probier ich gleich noch aus!

**Reply by schnuller123** [reply]
habs eingetragen und werds loggen
meld mich morgen wieder 
danke

**Reply by SybexX** [reply]
meine Überlegung, mit der Null als marker war:
- bis die Zahl sich ändert, braucht es warscheinlich Jahre
- Sie ist nur in der Hauptanzeige vorhanden (bei Version 5.0 nicht vorhanden und beim Test ist dort eine 8)

Wenn es funktioniert wie gedacht, sollte man ja eigentlich schon mit der Version 4 beide Probleme beseitigen können.

**Reply by schnuller123** [reply]
mal wieder geniale idee
habs nun drin, werd mich melden, noch mit V5
willst du die V5 wegfallen lassen?
wird die neue Version dann übernommen?
ps lief mit V5 und h die ganze Nacht ohne error
